<b> LOGISTIC REGRESSION</b>

In [2]:
#1 import the required libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
#2 read the dataset

train=pd.read_csv('F:/VIT COLLEGE/2nd SEM/MACHINE LEARNING/LAB/titanic_data.csv') #titanic_data.csv
train.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [4]:
# fix the predictor

df=train[['Survived', 'Pclass', 'Sex', 'Age', 'Fare']]


In [5]:
#4 change male to 1 and female to 0 encoding

df["Sex"] = df["Sex"].apply(lambda sex:1 if sex=="male" else 0)

<ipython-input-5-cc8377c72f35>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["Sex"] = df["Sex"].apply(lambda sex:1 if sex=="male" else 0)


In [6]:
df["Sex"]

0      1
1      0
2      0
3      0
4      1
      ..
886    1
887    0
888    0
889    1
890    1
Name: Sex, Length: 891, dtype: int64

In [7]:
#5 handling missing values - Data Imputation

df.isnull().sum()



Survived      0
Pclass        0
Sex           0
Age         177
Fare          0
dtype: int64

In [8]:
df['Age']=df['Age'].fillna(df['Age'].median()) #Outliners


<ipython-input-8-53f78a676525>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Age']=df['Age'].fillna(df['Age'].median()) #Outliners


In [9]:
#6 take a look at the dataframe
df.head()

Survived  Pclass  Sex   Age     Fare
0         0       3    1  22.0   7.2500
1         1       1    0  38.0  71.2833
2         1       3    0  26.0   7.9250
3         1       1    0  35.0  53.1000
4         0       3    1  35.0   8.0500

In [10]:
#7 set the predictor(X) and Response(Y) variables

X= df.drop("Survived", axis=1)
Y= df["Survived"]

In [11]:
#8 Splitting into the training and test set
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X,Y, test_size=0.3, random_state=25)

In [12]:
#9 call the logisctic regression model

from sklearn.linear_model import LogisticRegression
logit = LogisticRegression()
logit.fit(X_train,Y_train)


LogisticRegression()

In [15]:
#10 Compute the predictions or Y hat

Y_pred =logit.predict(X_test)



In [16]:
#11 confusion matrix

from sklearn.metrics import confusion_matrix
confusion_matrix= confusion_matrix(Y_test,Y_pred)
confusion_matrix

array([[136,  29],
       [ 31,  72]], dtype=int64)

In [17]:
#12 accuracy score

from sklearn.metrics import accuracy_score
accuracy_score(Y_test, Y_pred)

0.7761194029850746

In [18]:
#13 classification report

from sklearn.metrics import classification_report
report = classification_report(Y_test, Y_pred)
print(report)


              precision    recall  f1-score   support

           0       0.81      0.82      0.82       165
           1       0.71      0.70      0.71       103

    accuracy                           0.78       268
   macro avg       0.76      0.76      0.76       268
weighted avg       0.78      0.78      0.78       268



<b>NAIVE BAYES CLASSIFIER</b>

In [19]:
#1 call the naive bayes model

from sklearn.naive_bayes import GaussianNB

In [20]:
gnb = GaussianNB()
Y_pred = gnb.fit(X_train, Y_train).predict(X_test)

In [21]:
#2 accuracy score

from sklearn.metrics import accuracy_score
accuracy_score(Y_test, Y_pred)

0.7686567164179104

<b>Conclusion</b>: And finally from this two model, I can conclude that Logistic Regression model gives better accuracy (77%) while Naive Bayes gives 76% of accuracy.